### DISCUSSION TITLE
Is 2.5.8 exception for interactive elements side by side with insufficient spacing and target size if they are inline (that is part of paragraph)?

### DISCUSSION THREAD

**Discussion by Ashritha-98** [w3c/wcag#5066] [Q&A] [answered]
>Inline: The target is in a sentence or its size is otherwise constrained by the line-height of non-target text

I have come across a scenario where there are multiple buttons located side by side which are part of the paragraph.

Insufficient spacing
![image](https://github.com/user-attachments/assets/382e0702-7f00-418f-8cdd-b1bb84d5c6d8)
Insufficient target size
![image](https://github.com/user-attachments/assets/67a77f28-7e31-4f5a-9ccb-7e745bf6d363)

In this scenario, these buttons do not have a sufficient target size of at least 24 X 24 CSS px and also do not have sufficient spacing. 

In this scenario, we have interactive elements besides each other along with the non-target text.
The query over here is whether should we take it as non-compliance under Success Criteria (SC) 2.5.8 as the user may activate the other interactive control accidentally due to insufficient target size and spacing even though they are part of paragraph.
Your insights and guidance on addressing this issue and ensuring compliance with SC 2.5.8 would be greatly appreciated.

**Comment by melaniephilipp** [comment]
If I understand the scenario correctly, the targets are failing because their width and/or spacing along their width is not adequate. If that is correct, the width of the targets is not "constrained by the line-height of non-target text" and so the Inline exception does not apply to the width of the targets.

**Comment by patrickhlauke** [comment, marked as answer]
@melaniephilipp the exception is

> The target is in a sentence *or* its size is otherwise constrained by the line-height of non-target text

(emphasis added)

so yes, those buttons are currently in a sentence, and fall under the exception

**Comment by melaniephilipp** [comment]
@patrickhlauke ohhhhh... are you saying that even though their failure is due to their width - which is not constrained by the line-height of non-target text they completely get a "pass" under the Inline exception because of the wording of the SC - even though their width isn't constrained by the line-height? Is that an unintended consequence?

**Comment by aparnapasi** [comment]
@patrickhlauke @melaniephilipp do we have a conclusion on this? I have the same concern as @melaniephilipp . Though they are part of the sentence, width of the elements next to each other may make it hard for the user to choose. The whole point of the SC would not be met in this case.. Awaiting your thoughts on this.. Thanks!
